src/gwt/src/org/rstudio/studio/client/workbench/ui/PaneManager.java

```diff
@@ -289,10 +287,20 @@ public PaneManager(Provider<MainSplitPanel> pSplitPanel,
 
       // get the widgets for the extra source columns to be displayed
       ArrayList<Widget> sourceColumns = new ArrayList<>();
-      if (sourceColumnManager_.getSize() > 1 && additionalSourceCount_ > 0)
+      additionalSourceCount_ = userPrefs_.panes().getGlobalValue().getAdditionalSourceColumns();
```

Comment:
This should be `getValue()` (`getGlobalValue()` ignores project values, which is technically irrelevant for this option now but could change in the near future)

Reply:
Fixed! Thanks for the explanation 👍